### PR TITLE
feat: DUP v2 handle unreliable predictions - Frontend

### PR DESCRIPTION
**Pull request by cmaddox5**
**Asana task**: [[DUP v2] Handle unreliable predictions (devops checkboxes)](https://app.asana.com/0/1185117109217413/1204081228664130/f)

Spec [link](https://www.notion.so/mbta-downtown-crossing/Departures-Widget-Specification-20da46cd70a44192a568e49ea47e09ac?pvs=4#cb82633f5d314bcfa8974b721d249bd1)

**Frontend only**

This PR adds the frontend component needed to render the new `no_data_section` in the `Departures` widget. This section is always just a route's pill, the text "Updates unavailable", and the `NoConnection` icon. If all sections are on a rotation are unavailable, the `NoData` component will render instead. 

Screenshot 2023-03-10 at 9 50 55 AM

### CODE REVIEW

File: assets/src/components/v2/dup/no_data.tsx [hunk 1]
```diff
@@ -8,15 +8,19 @@ import NormalHeader from "./normal_header";
...
+}
+
+const NoData: ComponentType<Props> = ({ include_header }) => {
   let stationName = useOutfrontStation() || "Transit information";
   stationName = REPLACEMENTS[stationName] || stationName;
 
   return (
     <div className="no-data__container">
-      <NormalHeader text={stationName} />
+      {include_header && <NormalHeader text={stationName} />}
```
Comment (hannahpurcell) [inline assets/src/components/v2/dup/no_data.tsx:23]: These changes set up the ability to provide `include_header` to no_data, but all the instances of it in `dup.tsx` are set to true, so were these changes needed?

File: assets/src/components/v2/dup/dup_free_text.tsx [hunk 2]
```diff
@@ -3,7 +3,8 @@ import _ from "lodash";
 
 import { classWithModifier, classWithModifiers, imagePath } from "Util/util";
 
-const pillIcons = ["red", "blue", "orange", "green", "silver"];
+const textPills = ["red", "blue", "orange", "green", "silver"];
+const iconPills = ["cr", "bus"];
```
Comment (hannahpurcell) [inline assets/src/components/v2/dup/dup_free_text.tsx:7]: Ohhhhhh phoo... I just noticed that this file is an almost-duplicate of another v2 `free_text.tsx` file. Would you be interested in consolidating? If not, can make a clean-up task. I shouldn't have introduced this `dup_free_text.tsx` file. My bad

File: assets/src/components/v2/dup/no_data.tsx (hunk 1 above)
Comment (cmaddox5) [reply assets/src/components/v2/dup/no_data.tsx:23]: It is there to prevent the `departures_no_data` widget from rendering a duplicate header. The slot for `departures_no_data` is `:main_content_x` which means that a header candidate will still be given to the frontend to render. In the error state you added in `dup.tsx`, that specific no data state cannot render a header without hardcoding one in the component. So having this conditional allows each no data state to render what it needs to.

File: assets/src/components/v2/dup/dup_free_text.tsx (hunk 2 above)
Comment (cmaddox5) [reply assets/src/components/v2/dup/dup_free_text.tsx]: I don't mind doing it. I think everything v2 is using the same one now. I tested some of the other apps (`ReconstructedTakeover` uses `FreeText`) and it looks good.

File: assets/src/components/v2/dup/no_data.tsx (hunk 1 above)
Comment (hannahpurcell) [reply assets/src/components/v2/dup/no_data.tsx:23]: Ohhh I see now. 💯

File: assets/src/components/v2/dup/dup_free_text.tsx (hunk 2 above)
Comment (cmaddox5) [reply assets/src/components/v2/dup/dup_free_text.tsx]: Correct. We do technically get alerts for predictions, but it would only render if `alert_is_inline?/1` is `true` and that is hardcoded to `false` for now. I think it was left in and hardcoded to `false` so bus shelter alerts could be revisited, but that didn't happen as soon as we thought.

**Review by hannahpurcell — APPROVED** [review]
Looks great! Sorry, didn't mean to hold out on approving.
